Sources/Datadog/RUM/RUMVitals/VitalRefreshRateReader.swift

```diff
@@ -53,25 +54,47 @@ internal class VitalRefreshRateReader: ContinuousVitalReader {
         }
     }
 
-    // MARK: - Private
+    // MARK: - Internal
 
     @objc
-    private func displayTick(link: CADisplayLink) {
-        if let lastTimestamp = self.lastFrameTimestamp {
-            let frameDuration = link.timestamp - lastTimestamp
-            let currentFPS = 1.0 / frameDuration
-
-            for publisher in valuePublishers {
-                publisher.mutateAsync { currentInfo in
-                    currentInfo.addSample(currentFPS)
-                }
+    internal func displayTick(link: CADisplayLink) {
+        guard let fps = framesPerSecond(provider: link) else {
+            return
+        }
+
+        for publisher in valuePublishers {
+            publisher.mutateAsync { currentInfo in
+                currentInfo.addSample(fps)
             }
         }
-        lastFrameTimestamp = link.timestamp
     }
 
+    internal func framesPerSecond(provider: FrameInfoProvider) -> Double? {
+        var fps: Double? = nil
+
+        if let lastFrameTimestamp = self.lastFrameTimestamp {
+            let currentFrameDuration = provider.currentFrameTimestamp - lastFrameTimestamp
+            let currentFPS = 1.0 / currentFrameDuration
```

Comment:
nit/ I wonder if this could end up with "division by 0" - so far it didn't, this logic was already there. However, because we're processing `CADisplayLink` values here, I think it could make sense to add sanity check as a principle (_arithmetics with external values_). In looks hard to make assumptions on these values in certain edge situations.

Reply:
that's a fair a call, I will add some logic around divide by 0 case.